python/dgl/nodeflow.py

```diff
@@ -244,7 +246,8 @@ def map_to_parent_nid(self, nid):
         Tensor
             The parent node id array.
         """
-        return self._node_mapping.tousertensor()[nid]
+        nid = utils.toindex(nid)
```

Comment:
the input nid should be a tenor. why do we need to convert it dgl tensor first?

Reply:
The reason is for expression `a[b]`, MX requires `a` and `b` to be on the same device. However, sometimes, the `nid` is a GPU tensor while `self._node_mapping` is always on CPU. Converting to index not only handles the case that `nid` is not a tensor, but also does a GPU to CPU copying when the next line calls `nid.tousertensor()`

Reply:
could you add comments to explain the context conversion? and explain to users that the output tensor is always in the CPU context.

Reply:
Yeah. I wonder where I should add those comments. Actually, I discussed with @jermainewang about this yesterday and we agreed that we should let user know that graph itself has context and currently it's always on CPU. So user need to be aware that all graph query will should take CPU tensor as input and return CPU tensor as output.

Reply:
I'd prefer not to be crystal clear about this since we are likely to improve this part in the near release cycle. I think it's ok for now as the behavior is the same as before (both the argument and return tensor for index are CPU only).